File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/photodata/gateway/greendao/database/migration/MigrationV5.java
package nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.database.migration;

import org.greenrobot.greendao.database.Database;

import nl.acidcats.tumblrlikes.db_impl_greendao.PhotoEntityDao;

/**
 * Created by stephan on 11/05/2017.
 */

public class MigrationV5 implements Migration {
    @Override
    public Integer getVersion() {
        return 5;
    }

    @Override
    public void runMigration(Database db) {

        /*
        * Added to PhotoEntity:
        *     private boolean isFavorite;

    private boolean isHidden;
        *
        * */

        db.execSQL("ALTER TABLE " + PhotoEntityDao.TABLENAME + " ADD COLUMN " + PhotoEntityDao.Properties.IsFavorite.columnName + " INTEGER");
        db.execSQL("ALTER TABLE " + PhotoEntityDao.TABLENAME + " ADD COLUMN " + PhotoEntityDao.Properties.IsHidden.columnName + " INTEGER");
    }
}

File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/photodata/gateway/greendao/database/migration/MigrationV5.kt
package nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.database.migration

import nl.acidcats.tumblrlikes.db_impl_greendao.PhotoEntityDao
import org.greenrobot.greendao.database.Database

/**
 * Created on 25/10/2018.
 */
class MigrationV5 : Migration {
    override val version: Int = 5

    override fun runMigration(db: Database) {
        db.execSQL("ALTER TABLE " + PhotoEntityDao.TABLENAME + " ADD COLUMN " + PhotoEntityDao.Properties.IsFavorite.columnName + " INTEGER")
        db.execSQL("ALTER TABLE " + PhotoEntityDao.TABLENAME + " ADD COLUMN " + PhotoEntityDao.Properties.IsHidden.columnName + " INTEGER")
    }
}